src/Main.elm

```diff
@@ -25,40 +106,219 @@ init =
 
 type Msg
     = NoOp
+    | SelectedTab Tab
+    | SettingsMsg DataRetention.Msg
+    | ContactDetailsMsg ContactDetails.Msg
+    | TagsMsg Tags.Msg
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
-update _ model =
-    ( model, Cmd.none )
+update msg model =
+    case model of
+        Valid state ->
+            let
+                ( state_, cmd ) =
+                    case msg of
+                        SelectedTab tab ->
+                            if tab == state.selectedTab then
+                                ( state, Cmd.none )
+
+                            else
+                                ( { state | selectedTab = tab, pageState = initState tab state.userGroup }, Cmd.none )
+
+                        SettingsMsg msg_ ->
```

Comment:
Following the indents in this `update` function gets a bit difficult.
No need to modify it, but can you think of ways to make it more readable?

Reply:
Fair enough!